### ISSUE TITLE
Parse failure on variadic function argument in postgres dialect

### ISSUE THREAD

**Issue by pgr0ss** [sqlfluff/sqlfluff#7103]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

sqlfluff fails to parse a function call which uses a variadic argument:

I created a file called `test.sql`:
```sql
SELECT id FROM pgledger_create_transfers(
    event_at => '2025-07-21T12:45:54.123Z',
    VARIADIC transfer_requests => ARRAY[
        (:'user2_usd_id',:'liquidity_usd_id', '10.00'),
        (:'liquidity_eur_id',:'user2_eur_id', '9.26')
    ]::TRANSFER_REQUEST []
);
```

And here is the parse failure:
```
> uvx sqlfluff@latest --version
Installed 18 packages in 44ms
sqlfluff, version 3.4.2

> uvx sqlfluff@latest lint --dialect postgres test.sql                                                                                                                               == [test.sql] FAIL
L:   1 | P:  41 |  PRS | Line 1, Position 41: Found unparsable section: "(\n
                       | event_at => '2025-07-21T12:45:54.1..."
WARNING: Parsing errors found and dialect is set to 'postgres'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### Expected Behaviour

This is a valid function call, so I expected the SQL to parse properly. 

### Observed Behaviour

See above.

### How to reproduce

See above for a minimal example.

This function call is also from a real project, which you can see here: 
https://github.com/pgr0ss/pgledger/blob/f7bd300e7bf3e9769a2d1370ea80f449a7823876/examples/multi-currency.sql#L56-L62

### Dialect

postgres

### Version

3.4.2

### Configuration

No config

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by franloza** [comment]
I opened a PR related to `VARIADIC` arguments a couple of months ago: https://github.com/sqlfluff/sqlfluff/pull/7002. I'm opening a PR to fix this issue.
